source/camera/SlopeWalk.h

```diff
@@ -0,0 +1,106 @@
+/*
+ *  Copyright (C) 2011-2015  OpenDungeons Team
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef SLOPEWALK_H_
+#define SLOPEWALK_H_
+
+#include "camera/DummyArrayClass.h"
+#include "utils/Vector3i.h"
+
+#include <deque>
+#include <array>
+#include <string>
+
+
+
+/*! \brief The Class SlopeWalk keeps data required to rasterize
+ *   one polygon made of game tiles. To do that it remembers two
+ *  sides of polygon ; and on left and right side we keep a list 
+ *  of Vertexes and Slopes -- the tangens of an angle between two 
+ *  polygons edges ( remember the school equation y = ax + b
+ *  We add also a bunch of auxilary functions to be able to 
+ *  actually trace the position of some abstract point travelling
+ *  top - down along polygon's edges.
+ */
+
+class SlopeWalk
+{
+
+    friend class CullingManager;
+
+private:
+
+    std::deque<long long> mRightSlopes;
```

Comment:
it would be better to use sized variables such as uint32_t when needed as long long and such are plateform dependant and may create unexpected bugs.